src/lib/hooks/useTokenList/index.ts

```diff
@@ -14,31 +14,34 @@ import { validateTokens } from './validateTokenList'
 
 export const DEFAULT_TOKEN_LIST = 'https://gateway.ipfs.io/ipns/tokens.uniswap.org'
 
-const chainTokenMapAtom = atom<ChainTokenMap | undefined>(undefined)
+const tokenListAtom = atom<{ list?: string | TokenInfo[]; map?: ChainTokenMap }>({})
 
 export function useIsTokenListLoaded() {
-  return Boolean(useAtomValue(chainTokenMapAtom))
+  return Boolean(useAtomValue(tokenListAtom).map)
 }
 
 export function useSyncTokenList(list: string | TokenInfo[] = DEFAULT_TOKEN_LIST): void {
-  const { chainId, library } = useActiveWeb3React()
-  const setChainTokenMap = useUpdateAtom(chainTokenMapAtom)
-
   // Error boundaries will not catch (non-rendering) async errors, but it should still be shown
   const [error, setError] = useState<Error>()
   if (error) throw error
 
+  const [tokenList, setTokenList] = useAtom(tokenListAtom)
+  useEffect(() => setTokenList({ list }), [list, setTokenList])
+
+  const { chainId, library } = useActiveWeb3React()
   const resolver = useCallback(
     (ensName: string) => {
       if (library && chainId === 1) {
-        // TODO(zzmp): Use network resolver when wallet is not on chainId === 1.
         return resolveENSContentHash(ensName, library)
       }
       throw new Error('Could not construct mainnet ENS resolver')
     },
     [chainId, library]
   )
+
   useEffect(() => {
+    if (tokenList.map) return
```

Comment:
maybe `Array.isArray` instead? checking for method presence feels funky

Reply:
oh, `.map` is not the method, it's a property w/ a `TokenMap` value. THAT's confusing af.